boa/contracts/vyper/vyper_contract.py

```diff
@@ -646,9 +647,11 @@ def module_t(self):
     def source_map(self):
         if self._source_map is None:
             with anchor_settings(self.compiler_data.settings):
-                _, self._source_map = compile_ir.assembly_to_evm(
-                    self.compiler_data.assembly_runtime
-                )
+                if self._address is None:
+                    assembly = self.compiler_data.assembly
+                else:
+                    assembly = self.compiler_data.assembly_runtime
+                _, self._source_map = compile_ir.assembly_to_evm(assembly)
```

Comment:
not sure this is right, since it saves the wrong source map forever (maybe the wrong source map is not reachable later though)

Reply:
i think it would be better to use the source map overrider context manager for when we call the init function